nixos/modules/services/web-apps/overleaf.nix

```diff
@@ -0,0 +1,194 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.overleaf;
+in
+with lib;
+
+{
+  meta.maintainers = with maintainers; [ julienmalka camillemndn ];
+
+  options.services.overleaf = {
+    enable = mkEnableOption (mdDoc ''Overleaf'');
+
+    enableRedis = mkOption {
+      type = types.bool;
+      default = true;
+      description = mdDoc ''Redis'';
+    };
+
+    mongoDB = mkOption {
+      type = types.str;
+      default = "mongodb";
+      example = "ferretdb";
+      description = mdDoc ''The type of MongoDB to enable. One of "mongodb", "ferretdb" or "none"'';
+    };
+
+    texliveScheme = mkOption {
```

Comment:
could we not just pass the proper texlive packages we exactly want? sounds easier than mandating using a certain path

Reply:
Done! Julien actually scolded me for this too :p